### PR TITLE
Update st_stuff.c

### PR DESCRIPTION
**Pull request by dirsors**
Fix the following:
- Doom v1.9 doesn't allow cheats in nightmare mode!
- idclev cheat bug - it wasn't working.
- Ouch Face bug.

### CODE REVIEW

**Comment by Wolf3s** [comment]
> Fix the following:
> 
> * Doom v1.9 doesn't allow cheats in nightmare mode!
> * idclev cheat bug - it wasn't working.
> * Ouch Face bug.

Aproved

File: src/st_stuff.c [hunk 1]
```diff
@@ -683,7 +683,7 @@ ST_Responder (event_t* ev)
       
       if (gamemode == commercial)
       {
-	epsd = 0;
+	epsd = 1;
```
Comment (Arawn-Davies) [inline src/st_stuff.c:686]: Could you explain what this does? I've seen other ports just use epsd = 0;
https://github.com/chocolate-doom/chocolate-doom/blob/e55356d93a1408ca9db9a8cc57ec243f9fc8f99a/src/doom/st_stuff.c#L622

File: src/st_stuff.c [hunk 2]
```diff
@@ -804,7 +804,7 @@ void ST_updateFaceWidget(void)
 	    // being attacked
 	    priority = 7;
 	    
-	    if (plyr->health - st_oldhealth > ST_MUCHPAIN)
+	    if (st_oldhealth - plyr->health > ST_MUCHPAIN)
```
Comment (Arawn-Davies) [inline src/st_stuff.c:807]: Approved 👍

File: src/st_stuff.c (hunk 1 above)
Comment (Arawn-Davies) [reply src/st_stuff.c:686]: Reviewed & Approved

**Review by Arawn-Davies — APPROVED** [review]

**Pull request merged**